OpenMC_CSGModelMSFR/MSFR_3D_keff.py

```diff
@@ -0,0 +1,156 @@
+
+# Box above and below HX
+box_top=openmc.ZPlane(z0=0.95*100, boundary_type='reflective', albedo=0.000001)
+box_bottom=openmc.ZPlane(z0=-0.95*100, boundary_type='reflective', albedo=0.000001)
+box_outside=openmc.ZCylinder(r=2.17*100, boundary_type='reflective', albedo=0.000001)
+box_inside=openmc.ZCylinder(r=1.935*100, boundary_type='reflective', albedo=0.000001)
+
+# Curve above the HX
+inner_curve_top=openmc.ZTorus(a=1.785*100,x0=0,y0=0,z0=0.95*100,b=0.15*100,c=0.15*100, boundary_type='reflective', albedo=0.000001)
+outer_curve_top=openmc.ZTorus(a=1.785*100,x0=0,y0=0,z0=0.95*100,b=0.385*100,c=0.385*100, boundary_type='reflective', albedo=0.000001)
+# use box_top as bottom of curve
+curve_to_slant_bound=openmc.ZCylinder(r=1.785*100)
+
+# Curve below the HX
+inner_curve_bottom=openmc.ZTorus(a=1.785*100,x0=0,y0=0,z0=-0.95*100,b=0.15*100,c=0.15*100, boundary_type='reflective', albedo=0.000001)
+outer_curve_bottom=openmc.ZTorus(a=1.785*100,x0=0,y0=0,z0=-0.95*100,b=0.385*100,c=0.385*100, boundary_type='reflective', albedo=0.000001)
+
+# Core planes
+top_wall=openmc.ZCone(x0=0,y0=0,z0=0.8*100,r2=11.56, boundary_type='reflective', albedo=0.666666)
+top_cone_midplane=openmc.ZPlane(z0=0.8*100)
+bottom_wall=openmc.ZCone(x0=0,y0=0,z0=-0.8*100,r2=11.56, boundary_type='reflective', albedo=0.666666)
+bottom_cone_midplane=openmc.ZPlane(z0=-0.8*100)
+core_outerCyl=openmc.ZTorus(a=2.32*100,x0=0,y0=0,z0=0,b=1.265*100,c=1.265*100, boundary_type='reflective', albedo=0.666666)
+# Small planes parallel to walls
+top_small=openmc.ZCone(x0=0,y0=0,z0=0.575*100,r2=11.56, boundary_type='reflective', albedo=0.666666)
+top_small_cone_midplane=openmc.ZPlane(z0=0.575*100)
+bottom_small=openmc.ZCone(x0=0,y0=0,z0=-0.575*100,r2=11.56, boundary_type='reflective', albedo=0.666666)
+bottom_small_cone_midplane=openmc.ZPlane(z0=-0.575*100)
+slant_to_core_plane=openmc.ZCylinder(r=1.585*100)
+midplane=openmc.ZPlane(z0=0)
+
+## Regions/Volumes ##
+HX_region = -HX_top & +HX_inside & -HX_outside & +HX_bottom 
+box_region = -box_top & +box_inside & -box_outside & +box_bottom
+curve_top_region = +inner_curve_top & -outer_curve_top & +curve_to_slant_bound & +box_top
+curve_bottom_region = +inner_curve_bottom & -outer_curve_bottom & +curve_to_slant_bound & -box_bottom
+slant_top_region = +slant_to_core_plane & -curve_to_slant_bound & (+top_wall | -top_cone_midplane) & (-top_small & +top_small_cone_midplane) & +midplane
+slant_bottom_region = +slant_to_core_plane & -curve_to_slant_bound & (+bottom_wall | +bottom_cone_midplane) & (-bottom_small & -bottom_small_cone_midplane) & -midplane
+core_region=(+top_wall | -top_cone_midplane) & (+bottom_wall | +bottom_cone_midplane) & -slant_to_core_plane & +core_outerCyl
+# full_region = HX_region & box_region & core_region & curve_top_region & slant_top_region & curve_bottom_region & slant_bottom_region
+full_region = HX_region | box_region | core_region | curve_top_region | slant_top_region | curve_bottom_region | slant_bottom_region
```

Comment:
this is a little complicated (though likely necessary). maybe we add a readme that shows the different regions in a diagram?

Reply:
Adding an image of the geometry to the repository along with the GeN-Foam mesh it is supposed to represent